File: src/main/java/com/ansupercomputer/supercomputerhub/virtualmachine/VirtualMachine.java
package com.ansupercomputer.supercomputerhub.virtualmachine;

import com.ansupercomputer.supercomputerhub.user.User;

public class VirtualMachine {
    private int id;
    private int cores;
    private int storage;
    private int ram;
    private User owner;

    public VirtualMachine(int id, int cores, int storage, int ram, User owner) {
        this.id = id;
        this.cores = cores;
        this.storage = storage;
        this.ram = ram;
        this.owner = owner;
    }

    public int getId() {
        return id;
    }

    public int getCores() {
        return cores;
    }

    public int getStorage() {
        return storage;
    }

    public int getRam() {
        return ram;
    }

    public User getOwner() {
        return owner;
    }
}

File: src/main/java/com/ansupercomputer/supercomputerhub/virtualmachine/VirtualMachine.kt
package com.ansupercomputer.supercomputerhub.virtualmachine

import com.ansupercomputer.supercomputerhub.user.User

/**
 * Class that represents a Docker instance
 * @param id Docker instance id
 * @param cores How many cores the Docker instance is allowed to use
 * @param ram How much ram the Docker instance is allowed to use in GB
 * @param storage How much storage Docker instance is allowed to use in GB
 * @param owner Which owner the Docker instance belongs to
 */
class VirtualMachine (var id: Int, var cores: Int, var ram: Int, var storage: Int, var owner: User) {

}
